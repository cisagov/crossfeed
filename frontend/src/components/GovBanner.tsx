import React, { useState } from 'react';
import clsx from 'classnames';
import { makeStyles } from '@material-ui/core/styles';
import { ExpandLess, ExpandMore } from '@material-ui/icons';
import flagIcon from '../assets/us_flag_small.png';
import govIcon from '../assets/icon-dot-gov.svg';
import httpsIcon from '../assets/icon-https.svg';

const useStyles = makeStyles(theme => ({
  root: {
    position: 'relative',
    width: '100%',
    backgroundColor: '#f0f0f0',
    fontSize: '0.75rem',
    display: 'flex',
    alignItems: 'center'
  },
  inner: {
    width: '100%',
    display: 'flex',
    flexFlow: 'row nowrap',
    alignItems: 'center',
    maxWidth: 1440,
    margin: '0 auto',
    padding: `2px 24px`,
    [theme.breakpoints.down('sm')]: {
      padding: '2px 8px'
    }
  },
  flag: {
    padding: `0 ${theme.spacing()}px`
  },
  textWrap: {
    display: 'flex',
    flexFlow: 'row wrap',
    alignItems: 'center',
    flexGrow: 1,
    [theme.breakpoints.down('xs')]: {
      padding: `5px 0`
    }
  },
  text: {
    flex: '0 1 auto',
    marginRight: theme.spacing(),
    [theme.breakpoints.down('xs')]: {
      flex: '0 0 100%'
    }
  },
  btn: {
    border: 'none',
    background: 'none',
    padding: 0,
    outline: 'none'
  },
  btnExpand: {
    display: 'flex',
    alignItems: 'center',
    color: '#005ea2',
    textDecoration: 'underline'
  },
  infoInner: {
    maxWidth: 800,
    margin: '0 auto',
    lineHeight: 1.5,
    flexFlow: 'row wrap',
    padding: `${theme.spacing(3)}px 16px`
  },
  info: {
    display: 'flex',
    flex: '1 1 50%',
    padding: `${theme.spacing(2)}px ${theme.spacing(2)}px`,
    [theme.breakpoints.down('sm')]: {
      flex: '0 0 100%',
      padding: `${theme.spacing(2)}px 0`
    }
  },
  infoIcon: {
    minWidth: 40
  },
  infoText: {
    fontSize: '0.87rem',
    '& p': {
      margin: `0 ${theme.spacing(2)}px`
    }
  }
}));

export const GovBanner: React.FC = () => {
  const classes = useStyles();
  const [expanded, setExpanded] = useState(false);

  return (
    <>
      <div className={classes.root}>
        <div className={classes.inner}>
          <img src={flagIcon} alt="usa flag" className={classes.flag} />
          <div className={classes.textWrap}>
            <div className={classes.text}>
              An official website of the United States government
            </div>
            <button
              className={clsx(classes.text, classes.btn, classes.btnExpand)}
              onClick={() => setExpanded(exp => !exp)}
            >
              Here's how you know{' '}
              {expanded ? (
                <ExpandLess fontSize="small" />
              ) : (
                <ExpandMore fontSize="small" />
              )}
            </button>
          </div>
        </div>
      </div>
      {expanded && (
        <div className={classes.root}>
          <div className={clsx(classes.inner, classes.infoInner)}>
            <div className={classes.info}>
              <div className={classes.infoIcon}>
                <img src={govIcon} alt="Dot Gov" />
              </div>
              <div className={classes.infoText}>
                <p>
                  <strong>The .gov means it’s official.</strong>
                  <br />
                  Federal government websites often end in .gov or .mil. Before
                  sharing sensitive information, make sure you’re on a federal
                  government site.
                </p>
              </div>
            </div>
            <div className={classes.info}>
              <div className={classes.infoIcon}>
                <img src={httpsIcon} alt="HTTPS" />
              </div>
              <div className={classes.infoText}>
                <p>
                  <strong>This site is secure.</strong>
                  <br />
                  The <strong>https://</strong> ensures that you are connecting
                  to the official website and that any information you provide
                  is encrypted and transmitted securely.
                </p>
              </div>
            </div>
          </div>
        </div>
      )}
    </>
  );
};
