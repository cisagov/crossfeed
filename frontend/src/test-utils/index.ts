export * from './test-utils';
export * from './user';
export * from './organization';
