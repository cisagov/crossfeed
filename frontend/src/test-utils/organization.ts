import { Organization } from 'types';

export const testOrganization = {
  id: 'd7d6e913-0370-4f43-aebc-6bd727adc70e',
  name: 'test org',
  rootDomains: ['example.com'],
  ipBlocks: [],
  userRoles: [],
  scanTasks: [],
  isPassive: false,
  inviteOnly: false,
  granularScans: []
};
