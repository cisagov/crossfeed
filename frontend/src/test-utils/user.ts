import { AuthUser } from 'context';

export const testUser: AuthUser = {
  id: 'd7d6e913-0370-4f43-aebc-6bd727adc70e',
  createdAt: '2020-08-23T03:36:57.231Z',
  updatedAt: '2020-08-23T03:36:57.231Z',
  firstName: 'John',
  lastName: 'Smith',
  fullName: 'John Smith',
  invitePending: false,
  userType: 'standard',
  email: 'test@crossfeed.gov',
  roles: [],
  dateAcceptedTerms: new Date(),
  acceptedTermsVersion: 'v1-user',
  isRegistered: true
};
