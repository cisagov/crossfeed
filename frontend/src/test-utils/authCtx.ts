import { AuthContextType } from 'context';
import { testUser } from './user';

export const authCtx: AuthContextType = {
  user: testUser,
  token: 'some-test-token-never-verified',
  currentOrganization: null,
  loading: false,
  login: jest.fn(),
  logout: jest.fn(),
  setUser: jest.fn(),
  setOrganization: jest.fn(),
  refreshUser: jest.fn(),
  setLoading: jest.fn(),
  apiGet: jest.fn(),
  apiDelete: jest.fn(),
  apiPatch: jest.fn(),
  apiPost: jest.fn(),
  apiPut: jest.fn(),
  maximumRole: 'user',
  touVersion: 'v1-user',
  userMustSign: false
};
