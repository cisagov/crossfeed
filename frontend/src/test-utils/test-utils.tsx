import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { mocked } from 'ts-jest/utils';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext, AuthContextType } from '../context/AuthContext';
import { CFThemeProvider } from 'context';
import { authCtx } from './authCtx';

interface CustomRenderOptions extends RenderOptions {
  initialHistory?: string[];
  authContext?: Partial<AuthContextType>;
}

const customRender = (ui: any, options: CustomRenderOptions = {}) => {
  const { initialHistory, authContext, ...rest } = options;

  // Provide any context that the components may be expecting
  const Wrapper: React.FC = ({ children }) => (
    <CFThemeProvider>
      <MemoryRouter initialEntries={initialHistory}>
        <AuthContext.Provider
          value={{
            ...authCtx,
            ...authContext
          }}
        >
          {children}
        </AuthContext.Provider>
      </MemoryRouter>
    </CFThemeProvider>
  );
  return render(ui, { wrapper: Wrapper, ...rest });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, mocked };
