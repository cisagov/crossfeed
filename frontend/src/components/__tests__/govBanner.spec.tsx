import React from 'react';
import { GovBanner } from '../GovBanner';
import { render, fireEvent, wait } from 'test-utils/test-utils';

describe('Gov Banner Component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<GovBanner />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('expands and collapses as as expected', async () => {
    const { queryByText, getByText } = render(<GovBanner />);
    expect(queryByText(/Here\'s how you know/)).toBeInTheDocument();
    expect(queryByText('This site is secure.')).not.toBeInTheDocument();
    fireEvent.click(getByText(/Here\'s how you know/));
    await wait(() => {
      expect(getByText('This site is secure.')).toBeInTheDocument();
    });
    fireEvent.click(getByText(/Here\'s how you know/));
    await wait(() => {
      expect(queryByText('This site is secure.')).not.toBeInTheDocument();
    });
  });
});
