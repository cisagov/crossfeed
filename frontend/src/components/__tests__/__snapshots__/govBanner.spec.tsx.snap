// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gov Banner Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-inner-2"
    >
      <img
        alt="usa flag"
        class="makeStyles-flag-3"
        src="us_flag_small.png"
      />
      <div
        class="makeStyles-textWrap-4"
      >
        <div
          class="makeStyles-text-5"
        >
          An official website of the United States government
        </div>
        <button
          class="makeStyles-text-5 makeStyles-btn-6 makeStyles-btnExpand-7"
        >
          Here's how you know 
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
